plots/laws/thermodynamics/mass_of_ice_for_cooling_hot_water.py

```diff
@@ -0,0 +1,104 @@
+    energy_heating_law.temperature_end: temperature_balance
+})
+
+energy_to_heating_ice_equation = energy_heating_law.law.subs({
+    energy_heating_law.specific_heat_capacity: specific_heat_heating_ice,
+    energy_heating_law.body_mass: mass_of_ice,
+    energy_heating_law.temperature_origin: temperature_of_ice,
+    energy_heating_law.temperature_end: temperature_melt_ice
+})
+
+energy_to_melt_ice_equation = energy_melting_law.law.subs({
+    energy_melting_law.specific_heat_melting: specific_heat_melting_ice,
+    energy_melting_law.mass_of_matter: mass_of_ice
+})
+
+energy_to_heat_melted_ice_equation = energy_heating_law.law.subs({
+    energy_heating_law.specific_heat_capacity: specific_heat_heating_water,
+    energy_heating_law.body_mass: mass_of_ice,
+    energy_heating_law.temperature_origin: temperature_melt_ice,
+    energy_heating_law.temperature_end: temperature_balance
+})
+
+thermodinamics_law_1_equation = thermodinamics_law_1.law.subs({
+    thermodinamics_law_1.amounts_energy:
+    (energy_cooling_hot_water.rhs, energy_to_heating_ice_equation.rhs,
+    energy_to_melt_ice_equation.rhs, energy_to_heat_melted_ice_equation.rhs)
+}).doit()
+
+solve_system_equations = solve((thermodinamics_law_1_equation, mass_of_all_water_equation), (mass_of_ice, mass_of_hot_water))
+mass_of_ice_solve_value = solve_system_equations[mass_of_ice]
+mass_of_hot_water_solve_value = solve_system_equations[mass_of_hot_water]
+
+mass_ratio_value = simplify(mass_of_ice_solve_value / mass_of_hot_water_solve_value)
+print(f"Total equation:\n{print_expression(Eq(mass_of_ice / mass_of_hot_water, mass_ratio_value))}")
+
+mass_ratio_to_plot = mass_ratio_value.subs({
+    temperature_of_ice: to_kelvin(Celsius(-20)),
+    specific_heat_heating_water: 4200,      # joules / (kilogram * kelvin)
+    specific_heat_heating_ice: 2100,        # joules / (kilogram * kelvin)
+    specific_heat_melting_ice: 330_000,     # 1 kilojoules / kilogram = 10^3 joules / kilogram
+    temperature_melt_ice: to_kelvin(Celsius(0))
+})
+
+base_plot = plot(title="The mass of ice required to cool the hot water to a set temperature",
+    xlabel=r"$T_{balance}, K$",
+    ylabel=r"$m_{ice} / m_{hot_water}$",
+    backend=MatplotlibBackend,
+    legend=True,
+    show=False)
+
+for temperature_of_hot_water_value in temperature_of_hot_water_values:
+    mass_ratio_to_subplot = mass_ratio_to_plot.subs({
+        temperature_of_hot_water: to_kelvin(Celsius(temperature_of_hot_water_value))
+    })
+
+    # Find the upper limit of the temperature scale, so as not to build a graph
+    # on the temperature section where there will be a negative mass
+    temperature_supremum = solve(mass_ratio_to_subplot.subs({mass_of_ice: 0}), temperature_balance,
+        dict=True)[0][temperature_balance]
+
+    subplot = plot(mass_ratio_to_subplot, (temperature_balance, to_kelvin(Celsius(0)), temperature_supremum),
+        label=r"$T_{hot-water}=" + f"{temperature_of_hot_water_value}" + ", K$",
```

Comment:
It seems you forgot to convert temperature to Kelvin. Check your results before submission.